src/expandableCalendar/__test__/index.spec.ts

```diff
@@ -0,0 +1,294 @@
+
+      // events
+      expect(onMonthChange).not.toHaveBeenCalled(); // not called anyways...
+    });
+  });
+
+  describe('Knob', () => {
+    beforeEach(() => {
+      driver.render();
+    });
+
+    it('should expand expandable header ', () => {
+      driver.toggleKnob();
+      jest.runAllTimers();
+
+      expect(driver.isCalendarExpanded()).toBe(true);
+    });
+
+    it('should day press close expandable header', () => {
+      driver.toggleKnob();
+      jest.runAllTimers();
+      driver.selectDay(dashedToday);
+      jest.runAllTimers();
+      expect(driver.isCalendarExpanded()).toBe(false);
+    });
+  });
+
+  describe('CalendarList updates', () => {
+    describe('Day Press', () => {
+      beforeEach(() => {
+        driver.render();
+      });
+      it('should day press update date', () => {
+        driver.selectDay(dashedToday);
+
+        // WeekCalendar
+        const weekCalendarProps = driver.getWeekCalendar().props;
+        const initialScrollIndex = weekCalendarProps.initialScrollIndex;
+        const currentWeek = weekCalendarProps.data[initialScrollIndex];
+        expect(currentWeek).toBe(dashedToday);
+
+        // events
+        expect(onDateChanged).toHaveBeenCalled();
+        // expect(onMonthChange).toHaveBeenCalled();
+      });
+    });
+
+    describe('arrow press', () => {
+      beforeEach(() => {
+        driver.render();
+      });
+      it.each([[Direction.LEFT],[Direction.RIGHT]])(`should call onDateChanged and onMonthChanged to next month first day when pressing the %s arrow`, (direction: Direction) => {
+        driver.toggleKnob();
+        jest.runAllTimers();
+        const expectedDate = today.clone().addMonths(direction === Direction.RIGHT ? 1 : -1).setDate(1);
+        driver.pressOnHeaderArrow({left: direction === Direction.LEFT});
+        expect(onDateChanged).toHaveBeenCalledWith(toMarkingFormat(expectedDate), UpdateSources.PAGE_SCROLL);
+        expect(onMonthChange).toHaveBeenCalledWith(xdateToData(expectedDate), UpdateSources.PAGE_SCROLL);
+      });
+
+      it(`should call onDateChanged and onMonthChanged for first day in current month when changing to current month`, () => {
+        driver.toggleKnob();
+        jest.runAllTimers();
+        driver.pressOnHeaderArrow({left: false});
+        jest.runAllTimers();
+        driver.pressOnHeaderArrow({left: true});
+        jest.runAllTimers();
+        const expectedDate = today.clone().setDate(1);
+        expect(onDateChanged).toHaveBeenNthCalledWith(2, toMarkingFormat(expectedDate), UpdateSources.PAGE_SCROLL);
+        expect(onMonthChange).toHaveBeenNthCalledWith(2, xdateToData(expectedDate), UpdateSources.PAGE_SCROLL);
+      });
+
+      it(`should navigate 6 months ahead and back successfully`, () => {
+        driver.toggleKnob();
+        jest.runAllTimers();
+        times(6, () => {
+          driver.pressOnHeaderArrow({left: false});
+          jest.runAllTimers();
```

Comment:
Do we need to run all times after every press? Can't we move it to the end of the loop?

Reply:
fixed